test/widgets/cards_test.dart

```diff
@@ -0,0 +1,344 @@
+        expect(confimrButton, findsOneWidget);
+        await tester.tap(confimrButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byType(SnackBar), findsOneWidget);
+        verifyRequest(
+          processor,
+          urlMatcher: equals('/rest/v1/rpc/reject_ride'),
+          methodMatcher: equals('POST'),
+          bodyMatcher: equals({'ride_id': ride.id}),
+        ).called(1);
+      });
+
+      testWidgets('can abort rejecting Ride', (WidgetTester tester) async {
+        await openRejectDialog(tester);
+
+        //abort dialog
+        final Finder cancelButton = find.byKey(const Key('rejectCancelButton'));
+        expect(cancelButton, findsOneWidget);
+        await tester.tap(cancelButton);
+
+        //load page
+        await tester.pumpAndSettle();
+
+        expect(find.byKey(const Key('rejectSuccessSnackbar')), findsNothing);
+        verifyRequestNever(processor, urlMatcher: equals('/rest/v1/rpc/reject_ride'));
+      });
+    });
+  });
+
+  group('RideCard', () {
+    Future<void> loadRideCard(WidgetTester tester, List<ProfileFeature>? features) async {
+      ride = RideFactory().generateFake(
+        drive: NullableParameter(
+          DriveFactory().generateFake(
+            driver: NullableParameter(
+              ProfileFactory().generateFake(profileFeatures: features),
+            ),
+          ),
+        ),
+      );
+      whenRequest(processor).thenReturnJson(ride.drive!.toJsonForApi());
+      await pumpMaterial(tester, RideCard(ride));
+      //wait for card to load
+      await tester.pump();
+    }
+
+    testWidgets('shows the correct information', (WidgetTester tester) async {
+      await loadRideCard(tester, null);
+
+      //Profile Widget from driver is shown
+      final Finder profileWidgetFinder = find.byType(ProfileWidget);
+      expect(profileWidgetFinder, findsOneWidget);
+      final ProfileWidget profileWidget = tester.widget(profileWidgetFinder);
+      expect(profileWidget.profile, ride.drive!.driver);
+
+      //Rating is shown
+      expect(find.byType(CustomRatingBarIndicator), findsOneWidget);
+
+      //Price is shown
+      expect(find.byKey(const Key('price')), findsOneWidget);
+    });
+
+    group('Profile Features', () {
+      testWidgets('shows only first 3 Features', (WidgetTester tester) async {
+        await loadRideCard(tester, [
+          ProfileFeatureFactory().generateFake(rank: 1, feature: Feature.accessible),
+          ProfileFeatureFactory().generateFake(rank: 2, feature: Feature.noSmoking),
+          ProfileFeatureFactory().generateFake(rank: 3, feature: Feature.noVaping),
+          ProfileFeatureFactory().generateFake(rank: 4, feature: Feature.noPetsAllowed),
+        ]);
+
+        expect(find.byIcon(Icons.accessibility), findsOneWidget);
+        expect(find.byIcon(Icons.smoke_free), findsOneWidget);
+        expect(find.byIcon(Icons.vape_free), findsOneWidget);
+
```

Comment:
there should be a test that there is no `pets` icon

Reply:
thanks i had that not sure why i removed it xD